packages/sdk-effects/iframe-manager/src/lib/iframe-manager.effects.ts

```diff
@@ -0,0 +1,148 @@
+ * This follows the functional effect pattern, returning an API
+ * within a closure. Configuration could be passed here if needed
+ * for default behaviors (e.g., default timeout), but currently,
+ * all config is per-request.
+ *
+ * @returns An object containing the API for managing iframe requests.
+ */
+export default function iframeManagerInit(/*config: OAuthConfig*/) {
+  /**
+   * Creates an iframe to make a background request to the specified URL,
+   * waits for a redirect, and resolves with the query parameters from the
+   * redirect URL.
+   *
+   * @param options - The options for the iframe request (URL, timeout).
+   * @returns A Promise that resolves with the query parameters from the redirect URL or rejects on timeout or error.
+   */
+  const getParamsFromIFrame = (options: {
+    urlParams: string[];
+    url: AuthorizeUrl;
+    requestTimeout: number;
+  }): Promise<ResolvedParams<string>> => {
+    const { url, requestTimeout, urlParams } = options;
+
+    return new Promise((resolve, reject) => {
+      let iframe: HTMLIFrameElement | null = document.createElement('iframe');
+      let timerId: number | ReturnType<typeof setTimeout> | null = null;
+
+      // Define these within the promise closure to avoid retaining
+      // references after completion.
+      let onLoadHandler: () => void = () => {};
+      let cleanup: Noop = () => {};
+
+      cleanup = (): void => {
+        if (timerId) {
+          clearTimeout(timerId);
+          timerId = null;
+        }
+        if (iframe) {
+          iframe.removeEventListener('load', onLoadHandler);
+          // Check if iframe is still mounted before removing
+          if (iframe.parentNode) {
+            iframe.remove();
+          }
+          iframe = null; // Dereference iframe for garbage collection
+        }
+        onLoadHandler = () => {};
+        cleanup = () => {};
+      };
+
+      onLoadHandler = (): void => {
+        try {
+          // Accessing contentWindow or contentDocument can throw cross-origin errors
+          // if the iframe navigated to a different origin unexpectedly before redirecting back.
+          // We expect the final navigation to be back to the original redirect_uri origin.
+          if (iframe && iframe.contentWindow) {
+            const newHref = iframe.contentWindow.location.href;
+
+            const redirectUrl = new URL(newHref);
+            const params: ResolvedParams<string> = {};
+            redirectUrl.searchParams.forEach((value, key) => {
+              /**
+               * We want to match url params in the url,
+               * to params being passed into the iframe manager
+               * so we will ignore params that we are not explicitly
+               * asked to parse
+               */
+              if (urlParams.includes(key)) {
+                params[key] = value;
+              }
+            });
+
+            // Check for standard OAuth error parameters
+            if (params.error) {
```

Comment:
This block is still quite tied to OAuth related stuff. Let's add error params to the config parameter too. In other words, let's have an array of success params, and an array of error params. That way, this function doesn't need to know how to do anything other than the following:

1. Does this URL have any of the provided success params -> **_resolve_** with all params as object
2. Does this URL have any of the provided error params -> **_reject_** with all params as object